### PR TITLE
added some tests, minor clean-up

### PR DESCRIPTION
**Pull request by NicoWeidmann**
I added some tests that ensure a certain level of efficiency of the MaximumDeltaMatcher. Also, I removed some unused imports.
Additionally, I edited one test  in MoveFlipDownTest where I think an illegal position was expected when testing Move#getAffectedPositions.

Sorry for the many commits. :(

### CODE REVIEW

File: Final2-Another Candy Crush Clone/edu/kit/informatik/matchthree/tests/MatchThreeBoardConstructorTest.java [hunk 1]
```diff
@@ -121,6 +121,7 @@ public void testBoardDimensionsException04() {
     }
 
     @Test(expected = TokenStringParseException.class)
+    @Ignore("throw new IliasPostException - maybe *insert random exception* is more applicable here?")
```
Comment (deermichel) [inline Final2-Another Candy Crush Clone/edu/kit/informatik/matchthree/tests/MatchThreeBoardConstructorTest.java:124]: may the force - ILIAS - be with you

File: Final2-Another Candy Crush Clone/edu/kit/informatik/matchthree/tests/MoveFlipDownTest.java [hunk 2]
```diff
@@ -73,12 +73,14 @@ public void applyAndReverse() {
     public void getAffectedPositions() {
 
         Board board = new MatchThreeBoard(Token.set("AB"), 3, 3);
-        Move flipDown = new MoveFactoryImplementation().flipDown(Position.at(0, 2));
-        
+        // changed initial position, as the move would have been illegal
+        // otherwise (or was this intended?)
```
Comment (deermichel) [inline Final2-Another Candy Crush Clone/edu/kit/informatik/matchthree/tests/MoveFlipDownTest.java:77]: you're right - wasn't intended; unclear if exceptions should be thrown if getAffectedPositions() would return positions which aren't on the board

Comment (NicoWeidmann) [reply Final2-Another Candy Crush Clone/edu/kit/informatik/matchthree/tests/MoveFlipDownTest.java:77]: personally, I understand this sentence
> Gibt alle Positionen auf dem Spielbrett board zurück, die von dem Zug betroffen sein werden, also verändert werden.

in the way that an empty set should be returned by an illegal move, as it doesn't change the board upon application.

Comment (deermichel) [reply Final2-Another Candy Crush Clone/edu/kit/informatik/matchthree/tests/MoveFlipDownTest.java:77]: we'll never know...

**Pull request merged**